keps/sig-scheduling/NNNN-pod-to-node-assignment/README.md

```diff
@@ -0,0 +1,808 @@
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+A user wants a Pod to be scheduled alone on a node to avoid them from contending
+on resources (e.g. claiming all CPU cores available on a node). In other words,
+resource intensive workloads are needed to be scheduled on separate nodes.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+This section aims to describe how will the feature look in the pod spec.
+Following are some of the options that could be considered.
+
+### Introduce a new API Field
+
+A new API filed `pod.spec.nodeAsssignmentPolicy` could be introduced to be used
+by the new filter. The value for the newly introduced field could be an enum
+that has one of the following values: PodPerNode, Default where default is the
+same as the current behaviour and PodPerNode would enforce 1:1 pod to node
+mapping during scheduling of a given pod.
+
+```yaml
+...
+spec:
+  nodeAssignmentPolicy: <string>
+```
+
+### Define a NodeSelector/NodeAffinity Constraint
+
+A nodeSelector/nodeAffinity constraint can be used to achieve this 1:1 mapping,
+something similar to:
+
+```yaml
+spec:
+  nodeSelector:
+    pod-per-node: true
+```
+
+Or
+
+```yaml
+affinity:
+  nodeAffinity:
+    requiredDuringSchedulingIgnoredDuringExecution:
+      nodeSelectorTerms:
+      - matchExpressions:
+        - key: pod-per-node
+          operator: Exists
+```
+
+The new filter will look for this selector/affinity rule and filter out nodes
+that has pods other than daemonset pods. However, `NodeAffinity` filter will
+still evaluate this selector, and it will be looking for the labels on the nodes
+as well.
+
+Accordingly, there are a of couple points that still needs discussion
+
+-   Should the NodeAffinity filter be changed to ignore the node labels for this
+    particular case (nodes will not be required to have a `pod-per-node` label?
+
+-   Should we support setting this in nodeAffiniy where we need to handle OR/AND
+    logic of affinity constraints?
+
+### Test Plan
```

Comment:
Mention something like:

Unit and integration tests verifying that
- a single pod is assigned to a node while allowing daemonsets and static pods.
- kubelet rejects a pod if it was assigned directly to a node already running a pod with `SingleExcludingDaemonsetAndStaticPods` policy
- the api is no-op if feature gate is turned off

Reply:
Should we also add benchmark test evaluating performance of cases where all nodes are valid/invalid in a large cluster? 

or would this be similar to existing benchmarks?